service/matching/matchingEngine.go

```diff
@@ -256,6 +256,8 @@ func (e *matchingEngineImpl) AddActivityTask(addRequest *m.AddActivityTaskReques
 	return tlMgr.AddTask(addRequest.Execution, taskInfo)
 }
 
+var errQueryBeforeFirstDecisionCompleted = errors.New("query cannot be handled before first decision task is processed, please retry later")
```

Comment:
We should use QueryFailedError and let user retry.

Reply:
as discussed offline, will keep this behavior to wait/retry on server.